### ISSUE TITLE
an error occured at ui-library-add-call-recording running

### ISSUE THREAD

**Issue by nasu38yen** [Azure-Samples/communication-services-javascript-quickstarts#203]
Trying Sample:  ui-library-add-call-recording.
Run it locally and when I click the start recording button
An error occurs at this point.
/api/startRecording
var startRecordingResponse = await callAutomationClient.GetCallRecording().StartAsync(recordingOptions).ConfigureAwait(false);

[2024-01-16T05:31:33.394Z] Executed 'startRecording' (Failed, Id=f8d195c3-a594-4542-9d76-13f136813fed, Duration=15ms)
[2024-01-16T05:31:33.398Z] System.Private.CoreLib: Exception while executing function: startRecording. System.Private.CoreLib: The input is not a valid Base-64 string as it contains a non-base 64 character , more than two padding characters, or an illegal character among the padding characters.

It seems like there is nothing wrong with the connection string passed to new CallAutomationClient, but what does the error mean?

**Comment by dmceachernmsft** [comment]
Hi @nasu38yen! Thanks for filing the issue! we'll take a look and get back to you shortly!

**Comment by dmceachernmsft** [comment]
Hi @nasu38yen! I chatted with someone on the team responsible for call recording and learned that the `serverCallId` is different from the callId. essentially what the error is saying is the call recording service doesn't think that the `serverCallId` is correct. 

This value is something that is retrieved from the Call object itself and not something that you pass in. I will do some more investigation on my end and let you know when I know what specifically is going wrong!

**Comment by dmceachernmsft** [comment]
Hi @nasu38yen We are unable to repro the error on our end. Can you please provide us with some information:
- Version of Communication Calling
- Version of Communication react
- Device OS and Browser version
If the issue persists please check your connection strings and the ID and Token that you set in the `Secrets.ts` file.

Also make sure that you are not adding any characters to the reference `serverCallId` as this is the value causing you issues.

Finally a last tip is to make sure that you are running the apps in the correct order and viewing the app through the proxy port: `4280` you can find these steps at the bottom of the [README.md](https://github.com/Azure-Samples/communication-services-javascript-quickstarts/tree/main/ui-library-add-call-recording)

**Comment by nasu38yen** [comment]
Thank you for confirmation.
First, about the ID and Token that I set in the Secrets.ts file.
Tokens are valid for only 24 hours, so we reissue them on the portal every day.
In the first place, if the Token has expired, the UI will remain at Initializing... and will not start.
And the recording start action definitely reaches /api/startRecording.
In startRecording, I tried logging the serverCallId.

[2024-01-19T00:36:12.168Z] Point1 for aHR0cHM6Ly9hcGkuZmxpZ2h0cHJveHkuc2t5cGUuY29tL2FwaS92Mi9jcC9jb252LWpwZWEtMDQtcHJvZC1ha3MuY29udi5za3lwZS5jb20vY2 9udi81blk3ZXY0bjZrT3RzV1BLRTVKYVlnP2k9MTAtNjAtMTctMzEmZT02Mzg0MDY5MDAxNzAxMTMyMTQ
[2024-01-19T00:36:12.171Z] Point2 connection: endpoint=https://meet4u.communication.azure.com/;accesskey=Cjc+twVQPnKObBZHIZFcghVTdb3AiLUW8zIJbHnb2RdZJZl/jte3I9VKWqWt/GgQqXyLENzK0/iD4SR0i3 /TQ==

This is the version of the package you are using.

/api/api.csproj
     <PackageReference Include="Azure.Communication.CallAutomation" Version="1.0.0" />
     <PackageReference Include="Azure.Storage.Blobs" Version="12.10.0" />
     <PackageReference Include="Microsoft.Azure.EventGrid" Version="3.2.1" />
     <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="4.0.1" />

/app/package.json
     "@azure/communication-calling": "1.19.1-beta.2",
     "@azure/communication-calling-effects": "1.0.1",
     "@azure/communication-chat": "1.4.0-beta.2",
     "@azure/communication-react": "1.11.0-beta.1",

**Comment by dmceachernmsft** [comment]
Thank you for this information @nasu38yen! let me reach out to my colleagues on the Calling team about your `serverCallId` and see if what its returning is valid, I see a space in that id which might explain the error. So I will check with them what the expected formats here are.

**Comment by dmceachernmsft** [comment]
Hi @nasu38yen are you adding any custom code to the project at all? either in the `api` directory or in the client application? even if we force a space like the token we are seeing we do not get the same error that you are seeing. 

If you are adding anything are you okay with sharing it so we can look through what might be going on? 😊

**Comment by nasu38yen** [comment]
Thank you for your efforts.
No additions to the sample code.
I just edit /api/local.settings.json and /app/src/Secrets.ts and run it.
Also, I just added log code to /api/startRecording.cs for debugging.

OS: Windows 11 Pro Japanese version
Browser: Google Chrome 121.0.6167.86 (Official Build) (64bit)
Hardware: lenovo thinkpad x13 ntel(R) Core(TM) i5-10210U CPU @ 1.60GHz 2.11 GHz

**Comment by dmceachernmsft** [comment]
Thanks for the information @nasu38yen! Could you possible get us the `callId` of the call? That will help us figure out if its a bug from our calling SDK. You should be able to get this from pasting this snippet to line 56 - 64 in the `App.tsx` file in the app directory.
`newAdapter.onStateChange(async (state) => {
    if (state.call?.state === 'Connected') {
      const call = callAgent.calls.find((call) => call.id === state.call?.id);
      console.log(state.call.id);
      if (call) {
        setServerCallId(await call.info.getServerCallId());
      }
    }
  })`

**Comment by nasu38yen** [comment]
Thanks for the information @dmceachernmsft
callId is as follows
CallId:b0dc65bb-f463-4e53-b8ab-c3bf9b0e3b26

**Comment by dmceachernmsft** [comment]
@nasu38yen Thanks for the information I will check in with the Calling SDK team and get back to you!

**Comment by dmceachernmsft** [comment]
@nasu38yen still waiting to hear from our SDK team but they are looking into the issue 😊

**Comment by nasu38yen** [comment]
Additional information.
When I start the function, I see garbled characters in the terminal.
Do I need to make any settings related to character encoding when running Azure Functions Core Tools?
![image](https://github.com/Azure-Samples/communication-services-javascript-quickstarts/assets/4310058/ac26ccf1-5051-47f9-922c-8d57adcd8e75)

**Comment by nasu38yen** [comment]
I thought it was a problem that depended on my local development environment (for example, Japanese Windows), so I tried it on another PC (English Windows) and Github Codespaces, but the same error occurred on the API side when I started recording. .

**Comment by dmceachernmsft** [comment]
Thanks for the additional information @nasu38yen! Apologies for the late response another project took me away from this. I will reach out again to our calling team to see if there is any more information!

**Comment by dmceachernmsft** [comment]
@nasu38yen when you say that you see the same error in the when you start recording, do you mean the original error about the serverCallId? or the azure-function errors that you shared here?

**Comment by nasu38yen** [comment]
The error occurs at the following location in api/startRecording function.
var startRecordingResponse = await callAutomationClient.GetCallRecording().StartAsync(recordingOptions).ConfigureAwait(false);

[2024-02-21T23:49:04.597Z] Executed 'startRecording' (Failed, Id=64df1cd0-e030-4ce6-a232-0a0439f16561, Duration=123ms)
[2024-02-21T23:49:04.597Z] System.Private.CoreLib: Exception while executing function: startRecording. System.Private.CoreLib: The input is not a valid Base-64 string as it contains a non-base 64 character , more than two padding characters, or an illegal character among the padding characters.

I'm not sure if this error is based on serverCallId, but I've joined the call and the video is showing, so it seems like serverCallId is fine. An error occurs when I click the start recording button.

**Comment by nasu38yen** [comment]
Isn't it possible that an error occurred while decoding the accesskey of ACS_CONNECTION_STRING? I feel like this is the first time ACS_CONNECTION_STRING is used in a series of operations.

**Comment by dmceachernmsft** [comment]
Good news @nasu38yen we are able to find a repro. When pasting in your connection string, can you make sure that it has a suffix of `==` this denotes that the connection string is in base64. this should follow the `accesskey` variable of the connection string. the format should be:

`endpoint=<ENDPOINT_URL>;accesskey=<GUID_FOR_KEY>==`

we were able to repro be removing the `==` suffix.

let me know if this helps!

**Comment by nasu38yen** [comment]
I just copied the connection string from AzurePortal with == at the end and didn't remove it.
However, when I tried copying the connection string again today, the error in question did not occur and recording seemed to start.
[2024-02-26T02:08:03.008Z] Started recording for

I checked the connection string and there was no difference in the accesskey, but
The endpoint had changed.
aaaaaa.communication.azure.com→aaaaaa.japan.communication.azure.com
I don't think there was a japan subdomain before...
Anyway problem solved. thank you very much.
